### DISCUSSION TITLE
OpenID using Authelia started failing after working for weeks

### DISCUSSION THREAD

**Discussion by beastarman74** [outline/outline#3495] [Self Hosting] [answered]
I have all of a sudden stopped being able to login to Outline using OpenID authenication using Authelia. 

Authelia reports successful login:
time="2022-05-04T15:37:13+01:00" level=debug msg="Authorization Request with id '9462566a-66d3-4910-940e-4770ea36209f' on client with id 'outline1' is being processed" method=GET path=/api/oidc/authorization remote_ip=90.255.230.103
time="2022-05-04T15:37:13+01:00" level=debug msg="Authorization Request with id '9462566a-66d3-4910-940e-4770ea36209f' on client with id 'outline1' proceeding to lookup consent by challenge id '9373ba2d-cf6f-4937-a3bc-5a3793fb452e'" method=GET path=/api/oidc/authorization remote_ip=90.255.230.103
time="2022-05-04T15:37:13+01:00" level=debug msg="Authorization Request with id '9462566a-66d3-4910-940e-4770ea36209f' loaded consent session with id '144' and challenge id '9373ba2d-cf6f-4937-a3bc-5a3793fb452e' for client id 'outline1' and subject '422477e2-e6b3-4d66-8f12-83420fb38528' and scopes 'openid profile email groups'" method=GET path=/api/oidc/authorization remote_ip=90.255.230.103
time="2022-05-04T15:37:13+01:00" level=debug msg="Authorization Request with id '9462566a-66d3-4910-940e-4770ea36209f' on client with id 'outline1' was successfully processed, proceeding to build Authorization Response" method=GET path=/api/oidc/authorization remote_ip=90.255.230.103

 but Outline reports the following error:

{"error":"Failed to obtain access token","stack":"InternalOAuthError: Failed to obtain access token\n    at OAuth2Strategy._createOAuthError (/opt/outline/node_modules/passport-oauth2/lib/strategy.js:423:17)\n    at /opt/outline/node_modules/passport-oauth2/lib/strategy.js:177:45\n    at /opt/outline/node_modules/oauth/lib/oauth2.js:191:18\n    at ClientRequest.<anonymous> (/opt/outline/node_modules/oauth/lib/oauth2.js:162:5)\n    at ClientRequest.emit (node:events:526:28)\n    at ClientRequest.emit (node:domain:475:12)\n    at TLSSocket.socketErrorListener (node:_http_client:442:9)\n    at TLSSocket.emit (node:events:526:28)\n    at TLSSocket.emit (node:domain:475:12)\n    at emitErrorNT (node:internal/streams/destroy:157:8)\n    at emitErrorCloseNT (node:internal/streams/destroy:122:3)\n    at processTicksAndRejections (node:internal/process/task_queues:83:21)","level":"error","message":"Error during authentication"}

Does anyone have any ideas please?

**Comment by Limezy** [comment]
Do you remember changing or updating something on Outline or Authelia side ?

**Reply by beastarman74** [reply]
The only change I have made as far as I can remember is to change from using minio for emulated s3 storage locally to real amazon S3 storage (free tier) as I could not get it to be reliable state.

I have rolled back to an earlier configuration backup of Authelia but this hasn't fixed it.

**Reply by tommoor** [reply]
You can see the stack trace there, it goes all the way back to a failed network request in the [oauth](https://www.npmjs.com/package/oauth) library – it seems your auth server is not returning a successful response for some reason, this seems quite unlikely to be an issue in Outline.

**Reply by beastarman74** [reply]
I have been able to replicate the same error using Slack and Authelia OpenID.  I am working my way through the elements that make up my home lab (reverse proxy, adguard, cloudflare tunnel) trying to see if it is something in that config that is causing the issue.

Thanks for taking the time to look at this.

**Comment by beastarman74** [comment, marked as answer]
Its always DNS. It is ALWAYS DNS... Thanks @tommoor for the hint on network issue, it got me looking in the right place.

DNS settings on machine we set incorrectly.  Did recollect making a change but I thought I had made it on all my devices. Issue did not manifest straight away as I had cached authentication to Outline. Only when I logged out or tried a new device (Android Tablet) did I have the issue. 

Thanks again for the assistance. Hope I haven't wasted too much of the time.

**Reply by Zoranner** [reply]
If after checking it's confirmed that there are no DNS issues, and the OIDC debug information shows successful verification and returning of data, where else could the problem possibly be?

**Comment by Dorlas** [comment]
I ran into this problem in September, but only recently figured out how to solve it.
So - if you have a virtual machine, and there is a white address on it, 
then there are usually no problems with DNS.
BUT, if you have a gray address on your virtual machine, 
and you forward ports 80 and 443 through nat, this is where 
the problem with dns is.
The total solution that helped me already in 3 different configurations 
is to write all our white dns names (wiki, minio, keyloak, etc) 
in /ets/hosts and associate them with a real gray address physically assigned on the virtual machine.
The result - outside the address is white, and white addresses, 
but the virtual machine itself sees the dns addresses inside itself.
